### ISSUE TITLE
can you use {wk}?

### ISSUE THREAD

**Issue by mdsumner** [rnuske/apcf#7]
Just a note, see you working on wkb and wonder if you don't know about wk package?

that's all, best

**Comment by mdsumner** [comment]
also, I assume it's not of interest but {vapour} reads wkb by index or limit/skip with GDAL, there is {fastshp} by Simon Urbanek, and also https://github.com/paleolimbot/shp which might suit your needs

I'm generally interested in having access to the core tools here for development, it's crazy how spatial went from some sensible modularization to this intense GDAL-everywhere monolith ... but it's getting unpacked by various interested parties here and there.  Do you have discussions with other parties? I'm always interested in finding disparate efforts hopefully we can get more dev tools available for general use.

**Comment by rnuske** [comment]
Thanks for your comments and suggestions.

As you have seen the main idea is to get id of the huge GDAL dependency and the ridiculous IO Code I wrote because I couldn't figure out a better way to get deodata into my C++ Routines 2 years ago.

Following the newish Geo-R-stuff by you,  dcooley and paleolimbot I hoped I had gleaned enough ideas to make it work with less code and most importantly less dependencies. So the idea behind the WKB import is to have a minimum import routine without any dependencies besides GEOS. Something I could realize without depending on GDAL or any other R packages. This shall be complemented by accepting sfc in the best case without a hard dependency on sf similar to geos or sfheaders (maybe adding a dependency on wk, sfheaders or similar smallish packages). I would like to leave the import/export to the quadrillion geodata formats to packages which are better suited to handle the complexity of the different formats.
